src/services/chat/api.ts

```diff
@@ -0,0 +1,191 @@
+import firestore from '@react-native-firebase/firestore';
+import {
+  DocumentData, DocumentRefData, Query, QuerySnapshot,
+} from '../../types/services/Firebase';
+
+const chatDocument = (chatUID: string): DocumentRefData => firestore().collection('chats').doc(chatUID);
+
+const getChatByTarget = async (currentUser: DocumentRefData, targetUser: DocumentRefData): Promise<DocumentData | undefined> => {
+  const query = await firestore()
+    .collection('chats')
+    .where('users', 'in', [[currentUser, targetUser], [targetUser, currentUser]])
+    .get();
+
+  return query.docs.length > 0 ? query.docs[0] : undefined;
+};
+
+const createChat = async (users: Array<DocumentRefData>)
+  : Promise<DocumentRefData> => {
+  const { FieldValue } = firestore;
+
+  // First check if chat with those users already exists
+  const previousChat = await getChatByTarget(users[0], users[1]);
+  if (previousChat) {
+    return previousChat.ref;
+  }
+
+  const newChat = await firestore().collection('chats').add({
+    users,
+    updatedAt: FieldValue.serverTimestamp(),
+  });
+
+  await newChat.collection('updatedAt').add({ updatedAt: FieldValue.serverTimestamp() });
+
+  return newChat;
+};
+
+const getChat = async (chatUID: string): Promise<DocumentData> => firestore().collection('chats').doc(chatUID).get();
+
+const getOwnChats = (userRef: DocumentRefData)
+  : Query => firestore()
+  .collection('chats')
+  .where('users', 'array-contains', userRef);
+
+const latestMessageOnChat = async (
+  chatRef: DocumentRefData,
+): Promise<DocumentData> => new Promise((resolve, reject) => {
+  firestore()
+    .collection('messages')
+    .where('chat', '==', chatRef)
+    .orderBy('timestamp', 'desc')
+    .limit(1)
+    .get()
+    .then((querySnapshot) => {
+      resolve(querySnapshot.docs[0]);
+    })
+    .catch((err) => reject(err));
+});
+
+const loadMessages = async (
+  chatRef: DocumentRefData,
+  pageSize: number,
+  lastMessageTimestamp?: Date,
+): Promise<QuerySnapshot> => {
+  let query = firestore()
+    .collection('messages')
+    .where('chat', '==', chatRef)
+    .orderBy('timestamp', 'desc');
+
+  if (lastMessageTimestamp !== undefined) {
+    query = query.startAfter(lastMessageTimestamp);
+  }
+
+  return query.limit(pageSize).get();
+};
+
+const loadNewMessages = async (
```

Comment:
This looks similar enough to `LoadMessages` that, in the future, we could refactor both into a single function.